src/ontodia/diagram/defaultLinkRouter.ts

```diff
@@ -0,0 +1,218 @@
+            const element = diagramModel.getElement(sourceId);
+            const siblings = element.links.filter(link => {
+                const verts = link.get('vertices');
+                return link.get('source').id === link.get('target').id && (!verts || verts.length === 0);
+            });
+            const currentLinkIndex = siblings.indexOf(currentLink);
+
+            if (currentLinkIndex !== -1) {
+                vertices = updateFeedbackSibling(currentLink, element, currentLinkIndex);
+            }
+
+            let i = 0;
+            for (const link of siblings) {
+                if (link !== currentLink) {
+                    updateFeedbackSibling(link, element, i);
+                }
+                i++;
+            };
+        }
+
+        // Use the same direction for all siblings.
+        // We don't know whether the link goes from
+        // node A to node B or from B to A
+        // and to have it right directioned we define direction.
+        // In other words all links will have same direction because of the fact that
+        // all links will have the same couple of ids
+        const direction = sourceId > targetId;
+        if (direction) {
+            const temp = sourceId;
+            sourceId = targetId;
+            targetId = temp;
+        }
+
+        // If one of the ends is not a model, the link has no siblings.
+        if (!sourceId || !targetId) {
+            return vertices;
+        }
+
+        const links: Link[] = getLinksBetweenElements(diagramModel, sourceId, targetId);
+        if (links.length < 2) {
+            return vertices;
+        }
+
+        const siblings: Link[] = links.filter((link) => {
+            const verts = link.get('vertices');
+            if (verts && verts.length !== 0) {
+                return false;
+            }
+            const _sourceId = link.get('source').id;
+            const _targetId = link.get('target').id;
+
+            return _sourceId !== _targetId && (
+                  (_sourceId === sourceId && _targetId === targetId) ||
+                  (_sourceId === targetId && _targetId === sourceId)
+            );
+        });
+
+        // There is more than one siblings. We need to create vertices.
+        // First of all we'll find the middle point of the link.
+        const srcCenter = diagramModel.graph.getCell(sourceId).getBBox().center();
+        const trgCenter = diagramModel.graph.getCell(targetId).getBBox().center();
+        const midPoint = joint.g.line(srcCenter, trgCenter).midpoint();
+
+        // Then find the angle it forms.
+        const theta = srcCenter.theta(trgCenter);
+
+        // For mor beautifull positioning
+        const indexModifyer = siblings.length % 2 ? 0 : 1;
+
+        const currentLinkIndex = siblings.indexOf(currentLink);
+        if (currentLinkIndex !== -1) {
+            const v = updateSibling(currentLink, currentLinkIndex, siblings.length);
+            vertices = [v];
```

Comment:
Don't modify function parameters in the middle of the method, define a new variable instead.